external/storm-opentsdb/src/main/java/org/apache/stom/opentsdb/bolt/OpenTsdbBolt.java

```diff
@@ -0,0 +1,172 @@
+                for (OpenTsdbMetricDatapoint metricDataPoint : metricDataPoints) {
+                    metricPointsWithTuple.put(metricDataPoint, tuple);
+                }
+                batchHelper.addBatch(tuple);
+            }
+
+            if (batchHelper.shouldFlush()) {
+                LOG.debug("Sending metrics of size [{}]", metricPointsWithTuple.size());
+
+                ClientResponse.Details clientResponse = openTsdbClient.writeMetricPoints(metricPointsWithTuple.keySet());
+
+                if(failTupleForFailedMetrics && clientResponse.getFailed() > 0) {
+                    final List<ClientResponse.Details.Error> errors = clientResponse.getErrors();
+                    LOG.error("Some of the metric points failed with errors: [{}]", clientResponse);
+                    if(errors != null && !errors.isEmpty()) {
+
+                        Set<Tuple> failedTuples = new HashSet<>();
+                        for (ClientResponse.Details.Error error : errors) {
+                            final Tuple failedTuple = metricPointsWithTuple.get(error.getDatapoint());
+                            if(failedTuple != null) {
+                                failedTuples.add(failedTuple);
+                            }
+                        }
+
+                        for (Tuple batchedTuple : batchHelper.getBatchTuples()) {
+                            if(failedTuples.contains(batchedTuple)) {
+                                collector.fail(batchedTuple);
+                            } else {
+                                collector.ack(batchedTuple);
+                            }
+                        }
+
+                    } else {
+                        throw new RuntimeException("Some of the metric points failed with details: " + errors);
+                    }
+                } else {
+                    LOG.debug("Acknowledging batched tuples");
+                    batchHelper.ack();
+                }
+                metricPointsWithTuple.clear();
+            }
+        } catch (Exception e) {
+            batchHelper.fail(e);
+            metricPointsWithTuple.clear();
+        }
+    }
+
+    private List<OpenTsdbMetricDatapoint> getMetricPoints(Tuple tuple) {
+        List<OpenTsdbMetricDatapoint> metricDataPoints = new ArrayList<>();
+        for (TupleOpenTsdbDatapointMapper tupleOpenTsdbDatapointMapper : tupleOpenTsdbDatapointMappers) {
+            metricDataPoints.add(tupleOpenTsdbDatapointMapper.getMetricPoint(tuple));
+        }
+
+        return metricDataPoints;
+    }
+
+    @Override
+    public void cleanup() {
+        openTsdbClient.cleanup();
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer declarer) {
+        // this is a sink and no result to emit.
+    }
+
+    @Override
+    public Map<String, Object> getComponentConfiguration() {
+        return TupleUtils.putTickFrequencyIntoComponentConfig(null, flushIntervalInSeconds);
```

Comment:
We know that super.getComponentConfiguration() is null, but it would be more clear to state that explicitly.